src/main/g8/src/test/java/com/example/helloworld/GreeterTest.java

```diff
@@ -20,35 +21,34 @@
 
 public class GreeterTest {
 
-  private static ActorSystem serverSystem;
-  private static ActorSystem clientSystem;
+  // important to enable HTTP/2 in server ActorSystem's config
+  private static final Config config = ConfigFactory
+          .parseString("akka.http.server.preview.enable-http2 = on")
+          .withFallback(ConfigFactory.defaultApplication());
+  private static final ActorTestKit testKit = ActorTestKit.create(config);
+
+  private static ActorSystem<?> serverSystem = testKit.system();
+  private static ActorSystem<?> clientSystem;
   private static GreeterServiceClient client;
   
   @BeforeClass
   public static void setup() throws Exception {
-    // important to enable HTTP/2 in server ActorSystem's config
-    Config config = ConfigFactory.parseString("akka.http.server.preview.enable-http2 = on")
-        .withFallback(ConfigFactory.defaultApplication());
-    serverSystem = ActorSystem.create("HelloWorldServer", config);
     CompletionStage<ServerBinding> bound = new GreeterServer(serverSystem).run();
     // make sure server is bound before using client
     bound.toCompletableFuture().get(5, TimeUnit.SECONDS);
 
-    clientSystem = ActorSystem.create("HelloWorldClient");
-    Materializer clientMaterializer = ActorMaterializer.create(clientSystem);
+    clientSystem = ActorSystem.create(Behaviors.empty(), "GreeterClient");
     // the host and TLS certificate config are picked up from the config file
     client = GreeterServiceClient.create(
         GrpcClientSettings.fromConfig("helloworld.GreeterService", clientSystem),
-        clientMaterializer,
-        clientSystem.dispatcher()
+        clientSystem
       );
-
   }
 
   @AfterClass
   public static void teardown() {
-    TestKit.shutdownActorSystem(serverSystem);
-    TestKit.shutdownActorSystem(clientSystem);
+    TestKit.shutdownActorSystem(Adapter.toClassic(clientSystem));
+    testKit.shutdownTestKit();
```

Comment:
What's our recommendation when we need two actor systems in a test? This shutdown looks very inconsistent...

Reply:
I think it’s possible to have morethan one TestKitJunitResource

Reply:
Found `ActorTestkit.shutdown`.